enterprise-suite/scripts/lbc.py

```diff
@@ -0,0 +1,404 @@
+def execute(cmd, can_fail=False, print_to_stdout=False):
+    printerr(cmd)
+    if not args.dry_run:
+        stdout, returncode = run(cmd)
+        if print_to_stdout:
+            print(stdout)
+        else:
+            printerr(stdout)
+        if not can_fail and returncode != 0:
+            sys.exit("Command '" + cmd + "' failed!")
+        return returncode
+    return 0
+
+version_re = re.compile(r'([0-9]+\.[0-9]+(\.[0-9]+)?)')
+def require_version(cmd, required_version):
+    # Use first word as a program name in messages
+    name = cmd.partition(' ')[0]
+
+    # Use 1s timeout, mainly for docker when DOCKER_HOST is unreachable
+    stdout, returncode = run(cmd, 1)
+
+    if returncode == None:
+        sys.exit("Required program '" + name + "' not found")
+    elif returncode == 0 and stdout != '':
+        match = version_re.search(stdout)
+        if match != None:
+            current = LooseVersion(match.group())
+            required = LooseVersion(required_version)
+            if current >= required:
+                return
+            else:
+                sys.exit("Installed version of '" + name + "' is too old. Found: {}, required: {}"
+                    .format(current, required)) 
+
+    # Non-critical warning
+    printerr("warning: unable to determine installed version of '" + name + "'")
+
+def is_running_minikube():
+    stdout, returncode = run('minikube status')
+    return returncode == 0 \
+        and 'minikube: Running' in stdout \
+        and 'cluster: Running' in stdout
+
+def is_running_minishift():
+    stdout, returncode = run('minishift status')
+    return returncode == 0 \
+        and 'minishift: Running' in stdout \
+        and 'openshift: Running' in stdout
+
+# Helm check is a separate function because we also need it when not doing full
+# preflight check, eg. when using --export-yaml argument
+def check_helm():
+    require_version('helm version --client --short', REQ_VER_HELM)
+
+# Kubectl check is needed both in install and verify subcommands
+def check_kubectl(minishift=False):
+    require_version('kubectl version --client=true --short=true', REQ_VER_KUBECTL)
+
+    # Check if kubectl is connected to a cluster. If not connected, version query will timeout.
+    stdout, returncode = run('kubectl version', DEFAULT_TIMEOUT)
+    if returncode != 0:
+        msg = 'Cannot reach cluster with kubectl'
+        if minishift:
+            # Minishift needs special configuration for kubectl to work
+            msg = msg + ". Did you do 'eval $(minishift oc-env)'?"
+        sys.exit(msg)
+
+def preinstall_check(creds, minikube=False, minishift=False):
+    assert minikube == False or minishift == False, 'Did not expect both minikube and minishift running'
+
+    check_helm()
+    check_kubectl()
+
+    require_version("docker version -f '{{.Client.Version}}'", REQ_VER_DOCKER)
+
+    if minikube:
+        require_version('minikube version', REQ_VER_MINIKUBE)
```

Comment:
We shouldn't assume because minikube is running that we're deploying to it. You'd need to use `kubectl` to verify that (`kubectl config current-context` == `minikube`).

Reply:
Added kubectl context verification to the minikube check. I want to keep this particular version check because old minikube version was a problem I encountered myself about a week ago.